pandas/tests/plotting/test_converter.py

```diff
@@ -48,6 +48,7 @@ def test_registry_mpl_resets():
     code = (
         "import matplotlib.units as units; "
         "import matplotlib.dates as mdates; "
+        "import matplotlib.pyplot as plt; "
```

Comment:
This import is done in `pandas.io.formats.style` and changes what converters are registered with matplotlib. Adding Styler to the public API means this import is now done on `import pandas as pd`.

Another option here would be to determine `has_mpl = True` only if the `_mpl` context manager is used.